wasi/wasi_test.go

```diff
@@ -0,0 +1,261 @@
+package wasi
+
+import (
+	"encoding/binary"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/tetratelabs/wazero/wasm"
+)
+
+func TestWasiStringArray(t *testing.T) {
+	tests := []struct {
+		name            string
+		args            []string
```

Comment:
We should add a case where args == nil as I see the block `if arga == nil` is not covered

Reply:
Added it. Thanks!